server/src/main/java/keywhiz/utility/DSLContexts.java

```diff
@@ -40,12 +40,9 @@ public static DSLContext databaseAgnostic(DataSource dataSource) throws SQLExcep
     try (Connection conn = dataSource.getConnection()) {
       dialect = dialect(conn);
     }
-    if (dialect == SQLDialect.H2) {
-      return DSL.using(dataSource, SQLDialect.H2,
+    return DSL.using(dataSource, SQLDialect.H2,
```

Comment:
afaik, if you don't give `using` a dialect it calls `dialect(conn)` which "guesses" the dialect from the connection string.